core/modules/compare/store/actions.ts

```diff
@@ -1,38 +1,33 @@
-import Vue from 'vue'
 import { ActionTree } from 'vuex'
 import * as types from './mutation-types'
-import { htmlDecode } from '@vue-storefront/core/lib/store/filters'
-import i18n from '@vue-storefront/i18n'
-import rootStore from '@vue-storefront/core/store'
 import RootState from '@vue-storefront/core/types/RootState'
 import CompareState from '../types/CompareState'
-import { cacheStorage } from '../'
+import { StorageManager } from '@vue-storefront/core/lib/storage-manager'
 import { Logger } from '@vue-storefront/core/lib/logger'
+import { notifications } from '@vue-storefront/core/modules/compare/helpers'
+
 const actions: ActionTree<CompareState, RootState> = {
-  load ({ commit, getters }, force: boolean = false) {
+  async load ({ commit, getters, dispatch }, force: boolean = false) {
     if (!force && getters.isCompareLoaded) return
     commit(types.SET_COMPARE_LOADED)
-    cacheStorage.getItem('current-compare', (err, storedItems) => {
-      if (err) throw new Error(err)
+    const storedItems = await dispatch('getCurrentCompare')
+
+    if (storedItems) {
       commit(types.COMPARE_LOAD_COMPARE, storedItems)
       Logger.info('Compare state loaded from browser cache: ', 'cache', storedItems)()
-    })
+    }
+  },
+  async getCurrentCompare () {
```

Comment:
```suggestion
  async fetchCurrentCompare () {
```

fetch... for getting data and returning them; load... for getting data and modifying state :)